translator/translate/otel/receiver/otlp/translator.go

```diff
@@ -29,99 +34,161 @@ const (
 type translator struct {
 	common.NameProvider
 	common.IndexProvider
-	configKey string
-	signal    pipeline.Signal
-	factory   receiver.Factory
+	signal  pipeline.Signal
```

Comment:
Do we still need this field and the `WithSignal` function? It doesn't seem like they're used anymore since the `ParseOtlpConfig` function already finds the right config section.

Reply:
Nope, removed